src/tablet/tablet_impl.cc

```diff
@@ -5500,22 +5531,72 @@ void TabletImpl::DropFunction(RpcController* controller, const openmldb::api::Dr
+        PDLOG(WARNING, "pre-aggr table db path is not found. tid %u, pid %u", request->aggr_table_tid(), request->aggr_table_pid());
+        return;
+    }
+    std::string aggr_table_path = GetDBPath(aggr_path, request->aggr_table_tid(), request->aggr_table_pid());
+    if (!::openmldb::base::IsExists(aggr_table_path)) {
+        response->set_code(::openmldb::base::ReturnCode::kTableDbPathIsNotExist);
+        response->set_msg("pre-aggr table db path is not exist");
+        return;
+    }
+    std::string aggr_info;
+    google::protobuf::TextFormat::PrintToString(*request, &aggr_info);
+    std::string aggr_info_path = aggr_table_path + "/aggr_info.txt";
+
+    FILE* fd_write = fopen(aggr_info_path.c_str(), "w");
+    if (fd_write == NULL) {
+        PDLOG(WARNING, "fail to open file %s. err[%d: %s]", aggr_info_path.c_str(), errno, strerror(errno));
+    }
+    if (fputs(aggr_info.c_str(), fd_write) == EOF) {
+        PDLOG(WARNING, "write error. path[%s], err[%d: %s]", aggr_info_path.c_str(), errno, strerror(errno));
+        fclose(fd_write);
+    }
+    fclose(fd_write);
+    response->set_code(::openmldb::base::ReturnCode::kOk);
+    return;
+}
+
+bool TabletImpl::CreateAggregatorInternal(const ::openmldb::api::CreateAggregatorRequest* request,
+                                  std::string& msg) {
     const ::openmldb::api::TableMeta* base_meta = &request->base_table_meta();
     std::shared_ptr<Table> aggr_table = GetTable(request->aggr_table_tid(), request->aggr_table_pid());
     if (!aggr_table) {
-        response->set_code(::openmldb::base::ReturnCode::kTableIsNotExist);
-        response->set_msg("table is not exist");
         PDLOG(WARNING, "table is not exist. tid %u, pid %u", request->aggr_table_tid(), request->aggr_table_pid());
-        return;
+        msg.assign("table is not exist");
+        return false;
     }
+    auto aggr_replicator = GetReplicator(request->aggr_table_tid(), request->aggr_table_pid());
     auto aggregator = ::openmldb::storage::CreateAggregator(*base_meta, *aggr_table->GetTableMeta(),
-                                                            aggr_table, request->index_pos(),
+                                                            aggr_table, aggr_replicator, request->index_pos(),
                                                             request->aggr_col(), request->aggr_func(),
                                                             request->order_by_col(), request->bucket_size());
     if (!aggregator) {
-        response->set_code(::openmldb::base::ReturnCode::kError);
-        response->set_msg("create aggregator failed");
-        return;
+        msg.assign("create aggregator failed");
+        return false;
+    }
+
+    auto base_replicator = GetReplicator(base_meta->tid(), base_meta->pid());
+    if (!base_replicator) {
+        PDLOG(WARNING, "base table replicator is not exist. tid %u, pid %u", base_meta->tid(), base_meta->pid());
```

Comment:
return

Reply:
if base table is not been recovered, It's ok to continue. the replicator will be set when base table recovery.